packages/ui/src/components/RenameTag.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Alert,
+  Button,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { FunctionComponent, useState } from 'react';
+
+interface RenameTagProps {
+  isModalOpen: boolean;
+  currentName: string;
+  onClose: () => void;
+  onRename: (newName: string) => void;
+}
+
+export const RenameTag: FunctionComponent<RenameTagProps> = ({
+  isModalOpen,
+  currentName,
+  onRename,
+  onClose
+}) => {
+  type validate = 'success' | 'error' | 'default';
+
+  const [newName, setNewName] = useState<string>(currentName);
+  const [validated, setValidated] = useState<validate>('default');
+
+  const handleNameChange = (name: string) => {
+    setNewName(name);
+    const regex = /^[a-zA-Z0-9-]+$/;
```

Comment:
Creating RegExp is a expensive operation :thinking: , could we please create this RegExp once, and outside of the component definition? :pray: , otherwise we will be creating it every time the user clicks on the button.

Reply:
updated